src/main/kotlin/com/sourcegraph/cody/context/RemoteRepoInsight.kt

```diff
@@ -49,24 +44,48 @@ enum class RepoInclusion {
   MANUAL,
 }
 
+enum class RepoEnablement {
+  /** The user manually deselected the repository. */
+  DESELECTED,
+  /** Remote repo search did not find the repo (so it is disabled.) */
+  NOT_FOUND,
+  /** The repo has been found and is enabled. */
+  ENABLED,
+}
+
 data class RemoteRepo(
     val name: String,
-    var isEnabled: Boolean? = null,
-    val isIgnored: Boolean? = null,
-    val inclusion: RepoInclusion? = null
+    /**
+     * Null in the case of "not found" repos, or manually deselected repos we did not try to find.
+     */
+    val id: String?,
+    val enablement: RepoEnablement,
+    val isIgnored: Boolean,
+    val inclusion: RepoInclusion,
 ) {
+  val isEnabled: Boolean
```

Comment:
This naming conflict could be confusing. I'd probably call the RepoEnabelment `RepoCondition` and have the variants be `SELECTED`, `DESELECTED`, `NOT_FOUND`